pset/pset_test.go

```diff
@@ -63,3 +67,235 @@ func testRoundTrip(t *testing.T) {
+			t.Fatalf("Got: %s, expected: %s", hexRes, expectedHex)
+		}
+	}
+}
+
+func TestSigner(t *testing.T) {
+	file, err := ioutil.ReadFile("data/signer.json")
+	if err != nil {
+		t.Fatal(err)
+	}
+	var tests []map[string]interface{}
+	err = json.Unmarshal(file, &tests)
+
+	for _, v := range tests {
+		p, err := NewPsetFromBase64(v["base64"].(string))
+		if err != nil {
+			t.Fatal(err)
+		}
+		updater, err := NewUpdater(p)
+
+		for inIndex, vIn := range v["inputs"].([]interface{}) {
+			in := vIn.(map[string]interface{})
+			signature, _ := hex.DecodeString(in["signature"].(string))
+			pubkey, _ := hex.DecodeString(in["pubkey"].(string))
+			updater.Sign(inIndex, signature, pubkey, p.Inputs[inIndex].RedeemScript, p.Inputs[inIndex].WitnessScript)
+		}
+
+		base64Res, err := updater.Upsbt.ToBase64()
+		if err != nil {
+			t.Fatal(err)
+		}
+		hexRes, err := updater.Upsbt.ToHex()
+		if err != nil {
+			t.Fatal(err)
+		}
+		expectedBase64 := v["expectedBase64"].(string)
+		expectedHex := v["expectedHex"].(string)
+		if base64Res != expectedBase64 {
+			t.Fatalf("Got: %s, expected: %s", base64Res, expectedBase64)
+		}
+		if hexRes != expectedHex {
+			t.Fatalf("Got: %s, expected: %s", hexRes, expectedHex)
+		}
+	}
+}
+
+func TestFinalizer(t *testing.T) {
+	file, err := ioutil.ReadFile("data/finalizer.json")
+	if err != nil {
+		t.Fatal(err)
+	}
+	var tests []map[string]interface{}
+	err = json.Unmarshal(file, &tests)
+
+	for _, v := range tests {
+		p, err := NewPsetFromBase64(v["base64"].(string))
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		err = FinalizeAll(p)
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		base64Res, err := p.ToBase64()
+		if err != nil {
+			t.Fatal(err)
+		}
+		hexRes, err := p.ToHex()
+		if err != nil {
+			t.Fatal(err)
+		}
+		expectedBase64 := v["expectedBase64"].(string)
+		expectedHex := v["expectedHex"].(string)
+		if base64Res != expectedBase64 {
+			t.Fatalf("Got: %s, expected: %s", base64Res, expectedBase64)
+		}
+		if hexRes != expectedHex {
+			t.Fatalf("Got: %s, expected: %s", hexRes, expectedHex)
+		}
+	}
+}
+
+func TestExtractor(t *testing.T) {
+	file, err := ioutil.ReadFile("data/extractor.json")
+	if err != nil {
+		t.Fatal(err)
+	}
+	var tests []map[string]interface{}
+	err = json.Unmarshal(file, &tests)
+
+	for _, v := range tests {
+		p, err := NewPsetFromBase64(v["base64"].(string))
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		tx, err := Extract(p)
+		if err != nil {
+			t.Fatal(err)
+		}
+		res, err := tx.ToHex()
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		expectedTxHex := v["expectedTxHex"].(string)
+		if res != expectedTxHex {
+			t.Fatalf("Got: %s, expected: %s", res, expectedTxHex)
+		}
+	}
+}
+
+func toConfidentialValue(val int) ([]byte, error) {
+	unconfPrefix := byte(1)
+	b := bytes.NewBuffer([]byte{})
+	if err := transaction.BinarySerializer.PutUint64(b, binary.LittleEndian, uint64(val)); err != nil {
+		return nil, err
+	}
+	return append([]byte{unconfPrefix}, reverseBytes(b.Bytes())...), nil
+}
```

Comment:
Could you add an end to end test also? @altafan

Reply:
What do you mean by "adding an end"?

Reply:
e2e test. From creation to extraction in a single one-shot function?

Reply:
Also I noticed more failing test cases in the `psbt` repo. We'll add more coverage later?